### DISCUSSION TITLE
A page source code highlighter

### DISCUSSION THREAD

**Discussion by johanjanssens** [joomlatools/joomlatools-pages#668] [Snippets]
While working on the presentation for Joomladay USA with @woluweb I build a little template function to easily return the source code of a page. Useful when you use Pages to create your presentation. 

### Code

Here is the code for the function. Add it to: `/joomlatools-pages/extensions/pages/template/function/source.php`

```php
<?php
return function($page)
{
    $file   = $this->getObject('template.locator.factory')->locate('page://pages/'.$page);
    $source = file_get_contents($file);

    $parts = preg_split('#^\s*(---\s*[\s\S]*\s*---)\s*(.*)$#miU', $source, 2, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);

    $html = '';
    if(!empty($parts[0]))
    {
        $code = trim(htmlentities($parts[0]));
$html .= <<<CODE
<pre>
    <code>$code</code>
</pre>
CODE;

    }

    if(!empty($parts[1]))
    {
        $code = trim(htmlentities($parts[1]));
$html .= <<<CODE
<pre>
    <code>$code</code>
</pre>
CODE;

    }

    return $html;
};
```

### Use

Use it as follows:

```yaml
---
process:
   filters: [highlight]
---

<?= source('[path/to/page]'] ?>
```

`path/to/page`is the path of the page you want to get the source code for. To get nice code highlighting add the highlight filter. 

And thats a wrap!

**Comment by woluweb** [comment]
sooo handy ! :)
txs
